aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cpp

```diff
@@ -6,6 +6,7 @@
 #include <c10/cuda/CUDACachingAllocator.h>
 
 #include <cusparse.h>
+#include <ATen/native/cuda/linalg/CUDASolver.h>
```

Comment:
Why is this include needed?

Reply:
See https://github.com/pytorch/pytorch/pull/72306 - it creates `torch_cuda_linalg` as the only library depending on cudasolver and magma, so I move `CUDASolver.[cpp|h] header to `native/cuda/linalg/` folder

Reply:
This file doesn't seem to use anything from `<ATen/native/cuda/linalg/CUDASolver.h>` because it's cuSPARSE specific, not cuSOLVER.

Reply:
Good point, removed.